File: backend/src/main/java/com/voltpay/backend/repository/SettingsBackupRepository.java
package com.voltpay.backend.repository;

import com.voltpay.backend.model.SettingsBackup;
import org.springframework.data.mongodb.repository.MongoRepository;
import java.util.Optional;

public interface SettingsBackupRepository extends MongoRepository<SettingsBackup, String> {
    Optional<SettingsBackup> findByPhoneNumber(String phoneNumber);
}

File: backend/src/main/kotlin/com/voltpay/backend/repository/SettingsBackupRepository.kt
package com.voltpay.backend.repository

import com.voltpay.backend.model.SettingsBackup
import org.springframework.data.mongodb.repository.MongoRepository
import java.util.Optional

interface SettingsBackupRepository : MongoRepository<SettingsBackup, String> {
    fun findByPhoneNumber(phoneNumber: String): Optional<SettingsBackup>
}
